### PR TITLE
Add repo publishing

### PR DESCRIPTION
**Pull request by jinnatar**
This one, when finalized, will make it either go live or boom.

Related: https://github.com/kanidm/kanidm/issues/3078

TODO:
- [ ] Try out the deploy key warcrime: Would need to build on a different action, or see if https://github.com/crazy-max/ghaction-github-pages/issues/230 goes anywhere.
- [x] Test everything
- [ ] Rewrite this TODO into an actual deployment plan
- [x] Flip all the TODO values to real ones
- [x] Get the parent PR finalized
- [ ] Yeet this into main.
- [ ] If successful, come back and enable import & extend in the aptly action.

### CODE REVIEW

File: .github/workflows/create-apt-repo.yml [hunk 1]
```diff
@@ -115,21 +115,55 @@ jobs:
...
-  # Step 4. Publish the created repo if this meets the requirements for publishing
-  # i.e. need to ignore pull-requests and perhaps even merges that don't have a specific tag.
-  # Point being to avoid pushing out churn all the way to downstream users unless we actually want them to update.
+  # Step 4. Publish the created repo if and only if it's a push to main.
+  publish:
+    name: Deploy to GitHub Pages
+    needs: create-repo
+    runs-on: ubuntu-24.04
+    # TODO: Enable the conditional once PR testing is done
+    #if: github.ref == 'refs/heads/main'
```
Comment (yaleman) [inline .github/workflows/create-apt-repo.yml:142]: You could move this down to the "publish to PPA" step so it runs all the other bits regardless.

File: .github/workflows/create-apt-repo.yml [hunk 2]
```diff
@@ -16,13 +16,11 @@ jobs:
       fail-fast: false
       matrix:
         category:
-          # TODO: replace refs with real ones once PR is approved
           - name: stable
-            # ref: v1.4.0
-            ref: stable_deb_testing
+            ref: 1.3.0  # Previous stable branch + backport
```
Comment (yaleman) [inline .github/workflows/create-apt-repo.yml:20]: this should really be 1.3.3

**Review by yaleman — CHANGES_REQUESTED** [review]
minor nit but otherwise 🥳

**Comment by yaleman** [comment]
DEPLOY_PAT has been added

File: .github/workflows/create-apt-repo.yml (hunk 2 above)
Comment (jinnatar) [reply .github/workflows/create-apt-repo.yml:20]: So backporting on top of the v1.3.3 tag is way more annoying but 1.3.0 is a branch that currently sits at 1.3.3, so it sort of works. Future stables would indeed use a tag.

Comment (yaleman) [reply .github/workflows/create-apt-repo.yml:20]: Oh, joy. Fair. 😄

**Comment by jinnatar** [comment]
Latest run seems to be as expected: https://github.com/kanidm/kanidm_ppa_automation/actions/runs/11318699131?pr=4
.. i.e. builds succeed now for nightly, but fail for stable. I've created an example backport PR at https://github.com/kanidm/kanidm/pull/3108 which _should_ make stable buildable without affecting the `v1.3.3` tag.

But I could also just rip out the stable category now and launch with only nightly? I don't think it would kill the upgrade path for anyone.

**Comment by jinnatar** [comment]
Run is looking good so lets see what happens!

**Pull request merged**